src/nvim/window.c

```diff
@@ -3986,6 +3996,8 @@ static void enter_tabpage(tabpage_T *tp, buf_T *old_curbuf, int trigger_enter_au
   }
 
   redraw_all_later(NOT_VALID);
+
+  lastused_tabpage = old_curtab;
```

Comment:
`EVENT_TABENTER` event is handled on line 3993. shouldn't this be set before that?

Reply:
Yes!  Fixed and tested: https://github.com/neovim/neovim/pull/11626/files#diff-3bbc9305485da8705320941762a1d500R43-R50 .